### ISSUE TITLE
Menu links incorrectly limited to 255 characters

### ISSUE THREAD

**Issue by swirtSJW** [department-of-veterans-affairs/va.gov-cms#13278]
## Describe the defect
Slack reference https://dsva.slack.com/archives/CDHBKAL9W/p1681401208207779

Edited 5/23/23 to make this complete
When trying to update several urls in the megamenu. One of them keeps returning an error in Drupal.

For VA Leadership it should be this url: https://department.va.gov/biographies/?_bio_position=assistant-secretary%2Cchairman%2Cdeputy-secretary%2Cgeneral-counsel%2Cinspector-general%2Cprincipal-executive-director%2Csecretary%2Cunder-secretary%2Cva-chief-of-staff%2F%3Fredirect%3D1%2Cva-chief-of-staff

## To Reproduce
1. Go to '/admin/structure/menu/item/934/edit?destination=/admin/structure/menu/manage/header-megamenu'
2. paste the following url in the link field 'https://department.va.gov/biographies/?_bio_position=assistant-secretary%2Cchairman%2Cdeputy-secretary%2Cgeneral-counsel%2Cinspector-general%2Cprincipal-executive-director%2Csecretary%2Cunder-secretary%2Cva-chief-of-staff%2F%3Fredirect%3D1%2Cva-chief-of-staff'
3. Click save
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/e79301b3-347a-47ad-8391-b642f3acfd5d)
5. Go to ` /admin/reports/dblog` to see the  complete errors
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/422a6d52-32b9-40ad-824b-4e75839004d1)
```
Drupal\Core\Entity\EntityStorageException: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'url' at row 1: UPDATE "menu_tree" SET "menu_name"=:db_update_placeholder_0, "url"=:db_update_placeholder_1, "route_name"=:db_update_placeholder_2, "route_parameters"=:db_update_placeholder_3, "options"=:db_update_placeholder_4, "title"=:db_update_placeholder_5, "description"=:db_update_placeholder_6, "weight"=:db_update_placeholder_7, "enabled"=:db_update_placeholder_8, "expanded"=:db_update_placeholder_9, "parent"=:db_update_placeholder_10, "id"=:db_update_placeholder_11, "class"=:db_update_placeholder_12, "provider"=:db_update_placeholder_13, "metadata"=:db_update_placeholder_14, "form_class"=:db_update_placeholder_15, "has_children"=:db_update_placeholder_16, "discovered"=:db_update_placeholder_17, "depth"=:db_update_placeholder_18, "p1"=:db_update_placeholder_19, "p2"=:db_update_placeholder_20, "p3"=:db_update_placeholder_21, "p4"=:db_update_placeholder_22, "p5"=:db_update_placeholder_23, "p6"=:db_update_placeholder_24, "p7"=:db_update_placeholder_25, "p8"=:db_update_placeholder_26, "p9"=:db_update_placeholder_27, "route_param_key"=:db_update_placeholder_28 WHERE "mlid" = :db_condition_placeholder_0; Array ( [:db_update_placeholder_0] => header-megamenu [:db_update_placeholder_1] => https://department.va.gov/biographies/?_bio_position=assistant-secretary%2Cchairman%2Cdeputy-secretary%2Cgeneral-counsel%2Cinspector-general%2Cprincipal-executive-director%2Csecretary%2Cunder-secretary%2Cva-chief-of-staff%2F%3Fredirect%3D1%2Cva-chief-of-staff [:db_update_placeholder_2] => [:db_update_placeholder_3] => a:0:@ [:db_update_placeholder_4] => a:1:@s:8:"external";b:1; [:db_update_placeholder_5] => s:13:"VA Leadership"; [:db_update_placeholder_6] => [:db_update_placeholder_7] => 3 [:db_update_placeholder_8] => 1 [:db_update_placeholder_9] => 0 [:db_update_placeholder_10] => menu_link_content:2ff60fae-2cb7-4bb9-a2e5-c7fcf1c4bb59 [:db_update_placeholder_11] => menu_link_content:8a05f6c3-050b-4fa9-8db8-3dd33cccb468 [:db_update_placeholder_12] => Drupal\menu_link_content\Plugin\Menu\MenuLinkContent [:db_update_placeholder_13] => menu_link_content [:db_update_placeholder_14] => a:1:@s:9:"entity_id";s:3:"934"; [:db_update_placeholder_15] => \Drupal\menu_link_content\Form\MenuLinkContentForm [:db_update_placeholder_16] => 0 [:db_update_placeholder_17] => 0 [:db_update_placeholder_18] => 3 [:db_update_placeholder_19] => 1404 [:db_update_placeholder_20] => 1489 [:db_update_placeholder_21] => 1601 [:db_update_placeholder_22] => 0 [:db_update_placeholder_23] => 0 [:db_update_placeholder_24] => 0 [:db_update_placeholder_25] => 0 [:db_update_placeholder_26] => 0 [:db_update_placeholder_27] => 0 [:db_update_placeholder_28] => [:db_condition_placeholder_0] => 1601 ) in Drupal\Core\Entity\Sql\SqlContentEntityStorage->save() (line 815 of /var/www/cms/docroot/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php). [dd.trace_id=6067321073126737708 dd.span_id=1918468261515698340]
```
TLDR from @ndouglas  The actual error was that the URL was too long for the database field in which it was expected to be stored.

@dsasser Did some great research on this and discovered the likely problem. 

> There is a "hidden" 255 character limit here that is causing the error. The link uri field limit is 2048, whereas the menu_tree url field limit is 255. There is a core patch to increase the 255 limit to 2048. https://www.drupal.org/project/drupal/issues/3106205

## AC / Expected behavior
When I click save, even with a really long url, it saves.

## Additional context
Add any other context about the problem here. Reach out to the Product Managers to determine if it should be escalated as critical (**prevents users from accomplishing their work with no known workaround and needs to be addressed within 2 business days**).

5/23 edit
Swirt's thought's: an editor is blocked from doing her job with no known workaround.  In hindsight this should have been a critical defect

**Comment by swirtSJW** [comment]
@productmike  and @dawnpruitt  I just added the critical defect label because an editor is unable to do their job and there is no workaround for them to use.

I also added the CMS team label because menus in general are part of the drupal architecture.   Yes it happens to be surfacing in a Public websites product but it could have easilly come up on any of our hundreds of menus.  It is just that the megamenu is the first place where a link that long is in use.
